### PR TITLE
Add release management docs

### PR DESCRIPTION
**Pull request by alisterburt**
# Description
This PR adds documentation for our release timeline and introduces the concept of a release manager. This PR was based on [this discussion](https://github.com/napari/napari/discussions/4126#discussioncomment-2224614)

### CODE REVIEW

File: docs/developers/release.md [hunk 1]
```diff
@@ -4,10 +4,47 @@ This guide documents `napari`'s release process.
...
+`pip install --pre napari`
+
+# Release management
+The release will be coordinated by a release manager whose responsibilities include...
+
+## Two weeks before release (one week before release candidate)
+- Look through currently open PRs and get a sense of what would be good to merge before the first release candidate 
+- Create a zulip thread in the release channel letting people know the release candidate is coming and pointing out PRs that would be nice to merge before release
+
+At this stage, bug fixes should and features that are close to landing should be prioritised. The release manager will follow up with PR authors, reviewing and merging as needed.
```
Comment (DragaDoncila) [inline docs/developers/release.md]:
```suggestion
At this stage, bug fixes and features that are close to landing should be prioritized. The release manager will follow up with PR authors, reviewing and merging as needed.
```

I think we use `z`s and skip out on `u`s around here 😆 Just kidding but to be consistent we should use US spelling across the board

File: docs/developers/release.md [hunk 2]
```diff
@@ -4,10 +4,47 @@ This guide documents `napari`'s release process.
...
+  
+## Nine days before release (two days before release candidate)
+- Generate release notes with the script in the release folder
+- Fill in the release highlights and make a PR with the release notes
+
+At this point the release manager should really now be the only person merging PRs on the repo for the next week.
```
Comment (DragaDoncila) [inline docs/developers/release.md]:
```suggestion
At this point the release manager should ideally be the only person merging PRs on the repo for the next week.
```

File: docs/developers/release.md [hunk 3]
```diff
@@ -4,10 +4,47 @@ This guide documents `napari`'s release process.
...
+- Add any recently merged PRs to release notes
+- Merge release notes
+- Make the release candidate
+- Announce to release stream on zulip that the first release candidate is available for testing
+
+## The week before release
+- Merge any PRs and update release notes accordingly
+- Make new release candidates as necessary and announce them on zulip
+
+At this stage PRs merged should focus mainly on regressions and bug fixes. new features should wait until after release.
```
Comment (DragaDoncila) [inline docs/developers/release.md]:
```suggestion
At this stage PRs merged should focus mainly on regressions and bug fixes. New features should wait until after release.
```

File: docs/developers/release.md [hunk 4]
```diff
@@ -4,10 +4,47 @@ This guide documents `napari`'s release process.
 Currently, it only handles distribution, but as the project matures,
 it will include generating release notes, documentation, etc.
 
-This is mainly meant for the core developers who will actually be performing the release.
-They will need to have a [PyPI](https://pypi.org) account with upload permissions to the `napari` package.
+# Timeline
+New versions of `napari` will be released every two months. The first release candidate will be available one week prior to release for testing purposes. Multiple release candidates may become available during the week prior to release.
```
Comment (DragaDoncila) [inline docs/developers/release.md]: Do we want to actually say that the rc will become available first Tuesday of the month or no?

**Review by DragaDoncila — APPROVED** [review]
This looks good to me, I've made some minor suggestions for spelling/grammar etc. and just wondered whether we should mention that the rc is slated for first Tuesday of the month?

**Comment by DragaDoncila** [comment]
[Test failure](https://github.com/napari/napari/runs/5287003537?check_suite_focus=true#step:9:282) is because we download `cells3d` from `skimage` in two of our examples and we've been forbidden from doing so...?

File: docs/developers/release.md (hunk 4 above)
Comment (alisterburt) [reply docs/developers/release.md]: Good point! Will add when on a computer

**Comment by alisterburt** [comment]
Thanks for looking this over @DragaDoncila, and for pulling me up on my British-isms - I'll add the note about Tuesdays and merge later 🙂

File: docs/developers/release.md (hunk 4 above)
Comment (alisterburt) [inline docs/developers/release.md]:
```suggestion
New versions of `napari` will be released every two months. The first release candidate will be available one week prior to release for testing purposes. Multiple release candidates may become available during the week prior to release. Upcoming releases can be found in our public calendar.
```

**Review by jni — APPROVED** [review]

**Review by sofroniewn — APPROVED** [review]
Love this so much!! Thanks @alisterburt for turning this around so quick and thanks @DragaDoncila @jni for comments/ additions. I'm very excited to be bringing more folks into the release process 🚀

**Pull request merged**